packages/ember/tests/routing/query_params_test/shared_state_test.js

```diff
@@ -0,0 +1,88 @@
+import {
+  Controller,
+  Service
+} from 'ember-runtime';
+import Ember from 'ember';
+import { run } from 'ember-metal';
+import { jQuery } from 'ember-views';
+import { QueryParamTestCase, moduleFor } from 'internal-test-helpers';
+
+class SharedStateQPTestCase extends QueryParamTestCase {
```

Comment:
You can just define one class, instead of defining this one here only to extend it below.